src/fetchers/workspace-fetcher.js

```diff
@@ -0,0 +1,49 @@
+/* @flow */
+
+import type {PackageRemote, FetchedMetadata} from '../types.js';
+import type Config from '../config.js';
+import type {RegistryNames} from '../registries/index.js';
+
+export default class WorkspaceFetcher {
+  constructor(dest: string, remote: PackageRemote, config: Config) {
+    this.config = config;
+    this.dest = dest;
+    this.registry = remote.registry;
+    this.workspaceDir = remote.reference;
+  }
+
+  config: Config;
+  dest: string;
+  registry: RegistryNames;
+  workspaceDir: string;
+
+  setupMirrorFromCache(): Promise<?string> {
+    return Promise.resolve();
+  }
+
+  async fetch(defaultManifest: ?Object): Promise<FetchedMetadata> {
+    const pkg = await (async () => {
```

Comment:
My mistake, was looking at the BaseFetcher implementation. This can be `async` and I removed the IIFE.